File: 3001-3250/3228. Maximum Number of Operations to Move Ones to the End/solution.py
class Solution:
    def maxOperations(self, s: str) -> int:
        n = len(s)
        ans = 0
        prev = 0
        i = 0
        while i < n:
            if s[i] == "1":
                prev += 1
            else:
                ans += prev
                while i + 1 < n and s[i + 1] == "0":
                    i += 1
            i += 1
        return ans

File: 3001-3250/3228. Maximum Number of Operations to Move Ones to the End/solution.cpp
class Solution {
public:
    int maxOperations(string s) {
        int n = s.size() ; 
        int ans = 0 ; 
        int prev = 0 ; 
        for( int i=0 ; i<n ; ++i ){
            if( s[i] == '1' ) prev++ ; 
            else {
                ans += prev ; 
                while( i+1<n && s[i+1] == '0' ) i++ ;
            }
        }
        return ans;
    }
};